aks/locals.tf

```
@@ -1,6 +1,17 @@
 locals {
   use_managed_identity = try(var.metrics_storage.managed_identity_node_rg_name != null, false)
 
+  metrics_storage = var.metrics_storage != null ? {
+    storage_config = {
+      type = "AZURE"
+      config = merge({
+        container       = var.metrics_storage.container
+        storage_account = var.metrics_storage.storage_account
+        },
+      local.use_managed_identity ? null : { storage_account_key = var.metrics_storage.storage_account_key })
+    }
+  } : null
```

Strange, I tested my suggestion and terraform fmt agreed with it. But to make it even more readable, I'd actually suggest this syntax:
```suggestion
  metrics_storage = var.metrics_storage == null ? null : {
    storage_config = {
      type = "AZURE"
      config = merge(
        {
          container       = var.metrics_storage.container
          storage_account = var.metrics_storage.storage_account
        },
        local.use_managed_identity ? null : {
          storage_account_key = var.metrics_storage.storage_account_key
        }
      )
    }
  }
```

---

It's nice :wink: I've commited, could you reapprove?